lib/runner/variant-reassignment.js

```diff
@@ -373,23 +398,50 @@ export default class VariantReassignment {
   }
 
   /**
-   * Select products that has at least one variant from the productDraft.
+   * Select products that has at least one variant
+   * or at least one matching slug with the productDraft.
    * @param productDraft
-   * @param products
+   * @param products if present, then this will be used and no fetching from CTP is needed
    * @returns {*}
    * @private
    */
   _selectMatchingProducts (productDraft, products) {
     const productDraftSkus = this.productService.getProductDraftSkus(productDraft)
+    const productDraftSlugs = productDraft.slug
     if (products) {
-      const skuToProductMap = this._createSkuToProductMap(products)
-      const matchingProducts = productDraftSkus.map(sku => skuToProductMap.get(sku))
-
-      // When there are new variants in productDraft the matchingProducts will contain
-      // undefined values which we filter out using _.compact method
-      return _.uniq(_.compact(matchingProducts))
+      // select products that matches by at least one slug with productDraft
+      const matchingProductsBySlug = this._selectMatchingProductsBySlug(products, productDraft)
+      // select products that matches by at least one variant with productDraft
+      const matchingProductsBySku = this._selectMatchingProductsBySkus(products,
+        productDraftSkus)
+      return _.compact(_.uniq(matchingProductsBySku.concat(matchingProductsBySlug)))
```

Comment:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/45/commits/7de9abbbd99e7543f68125818830000ba695bb69

Reply:
Hello @lojzatran  .. thank you for your update. I would vote for using
```js
return _.uniqWith(
  _.compact(matchingProductsBySku.concat(matchingProductsBySlug)),
  (first, second) => first.id === second.id
)
```
Instead of:
```js
return _.compact(_.uniqWith(matchingProductsBySku.concat(matchingProductsBySlug), _.isEqual))
```

Because we first filter out `false` values so we won't have to compare them and then we compare products just by their IDs instead of doing a deep-equal on the whole object which is more complex and slower.

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/45/commits/1676fcc9e0b7666bffb7806b25a6fca53721461e